api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -180,20 +181,50 @@ type KonnectExtensionStatus struct {
 	// +kubebuilder:validation:Optional
 	DataPlaneClientAuth *DataPlaneClientAuthStatus `json:"dataPlaneClientAuth,omitempty"`
 
+	// Konnect contains the status information related to the Konnect Control Plane.
+	//
+	// +kubebuilder:validation:Optional
+	Konnect *KonnectExtensionControlPlaneStatus `json:"konnect,omitempty"`
+
 	// Conditions describe the current conditions of the KonnectExtensionStatus.
 	// Known condition types are:
 	//
-	// * "Accepted"
-	// * "SecretProvisioned"
-	// * "ControlPlaneRefValid"
-	//
 	// +optional
 	// +listType=map
 	// +listMapKey=type
 	// +kubebuilder:validation:MaxItems=8
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// ClusterType is the type of the Konnect Control Plane.
+type ClusterType string
+
+const (
+	// ClusterTypeControlPlane is the type of the Konnect Control Plane.
+	ClusterTypeControlPlane ClusterType = "ControlPlane"
+	// ClusterTypeK8sIngressController is the type of the Kubernetes Control Plane.
+	ClusterTypeK8sIngressController ClusterType = "K8SIngressController"
+)
+
+// KonnectExtensionControlPlaneStatus contains the Konnect Control Plane status information.
+type KonnectExtensionControlPlaneStatus struct {
```

Comment:
Given that all the fields in this struct are required: what will be the values of `controlPlaneID` and `endpoints` when `clusterType` is `K8SIngressController`? Those fields do not make much sense to me for the ingress controller use case.

Reply:
clusterType `K8sIngressController` is meant to be set when configuring KIC in Konnect (i.e., KIC pushes the configuration directly to Kong). All these values still make sense as they have proper values, as there is a ControlPlane entity in Konnect.